src/sensor.cpp

```diff
@@ -1047,6 +1073,16 @@ void log_callback_end( uint32_t fps,
 
         _hid_device->stop_capture();
         _is_streaming = false;
+
+        std::string device_name = this->get_device().shared_from_this()->get_info(RS2_CAMERA_INFO_NAME);
```

Comment:
Service func?

Reply:
we changed it to:
get_device().get_info(RS2_CAMERA_INFO_NAME)

do you think this should be reduced also and replaced with service func ? 
if so, give me an example for what you mean in service func

Reply:
I mean inside `aus`, maybe you can add functions like this in use it:

```
register_all_supported_streams(librealsense::device device)
{
     std::string device_name = device.get_info(RS2_CAMERA_INFO_NAME);
        auto active_streams = get_active_streams();
        for (auto profile : active_streams)
        {
            librealsense::aus_system_timer_stop(rs2_stream_to_string(profile->get_stream_type()), device_name);
        }
}
```

The main idea is when the user look at device creation code, we doesn't care about AUS, if it's a single line call he can ignore it.
If you place a 5 lines of code into a 10 line function it will take over and it is not the main logic there, should be unnoticeable as we can IMO

Agree?

Reply:
fixed